app/views/sales/index.html.erb

```diff
@@ -0,0 +1,14 @@
+<h1>My Sales</h1>
+
+<ul>
+  <% sales_total = [] %>
+  <% @all_sales.each do |sale| %>
+    <% if sale.vendor_id == @vendor.id %>
+      <li><%= sale.amount %> <%= Product.find(sale.product_id).name %></li>
+      <% sales_total << sale.amount %>
+    <% end %>
+  <% end %>
+  <h1>Total Sales: </h1><%= sales_total.reduce(0){|sum,x| sum + x } %></h1>
```

Comment:
I think this could also go inside a method in the controller, and then you can call the method here in the view.  That way you can re-use the method anywhere.